### PR TITLE
Bump up flutter to 3.16

### LINKED ISSUE #14: Bumping up flutter sdk version
The following error is observed on changing sdk variable in pubsec.yaml 

```Because every version of flutter_localizations from sdk depends on intl 0.18.0 and flutter_form_builder >=6.0.0-nullsafety.0 <7.0.0-alpha.0 depends on intl ^0.17.0, flutter_localizations from sdk is incompatible with flutter_form_builder >=6.0.0-nullsafety.0 <7.0.0-alpha.0.
So, because noticeboard depends on both flutter_form_builder ^6.1.0+1 and flutter_localizations from sdk, version solving failed.```

### PR DESCRIPTION
**Pull request by pranavkonidena**
# Description

This PR updates the flutter sdk version to >= 3.16, useMaterial3 flag has been set to false for now as migration to Material UI 3 has to be discussed with the design team

Fixes #14 

## Type of change
Dependency update

## Screenshot
Screenshot 2023-12-26 at 3 48 28 PM

### CODE REVIEW

File: noticeboard/ios/Flutter/AppFrameworkInfo.plist [hunk 1]
```diff
@@ -21,6 +21,6 @@
   <key>CFBundleVersion</key>
   <string>1.0</string>
   <key>MinimumOSVersion</key>
-  <string>9.0</string>
```
Comment (just-ary27) [inline noticeboard/ios/Flutter/AppFrameworkInfo.plist:24]: iOS version kyu increase ho rha? Try to run it on an emulator with iOS 9.0

File: noticeboard/ios/Podfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Uncomment this line to define a global platform for your project
- platform :ios, '10.0'
```
Comment (just-ary27) [inline noticeboard/ios/Podfile:2]: Same here

**Review by just-ary27 — CHANGES_REQUESTED** [review]

**Comment by just-ary27** [comment]
Also readme me ios link dalde :)

**Comment by pranavkonidena** [comment]
Xcode pe min IOS version 11 ke neeche dikha hi nhi rha option mei.. added readme ios link

**Pull request closed without merging**